packages/core/src/utils.ts

```diff
@@ -58,5 +61,16 @@ export default class Utils {
         }
     }
 
-
+    /**
+     * Gets the default anchoring service.
+     *
+     * The default service anchors on Ropsten. If it's not specified, return the in-memory one.
+     *
+     * Note: only the first anchor service is returned from the list. If there are multiple services for the same
```

Comment:
In that case, why keep a list at all?  Let's only store a single anchor service per chainId for now and simplify the code.  If/when we support multiple anchor services for the same chainId, we can change it then.

Reply:
Right now, anchor service can return a list of chain IDs which it supports. That's why it's a list.

Reply:
One anchor service can support multiple chainIds, but one js-ceramic instance can't support multiple anchor services for the *same* chainId

Reply:
We haven't defined that still. There will be more anchoring services on the network which can be passed to the `js-ceramic` instance. Some of them will support same chainIds but different anchoring intervals, reputation, etc. This way we can support more anchoring services and choose which one to use based on the other properties.

Reply:
There might be special case anchor services for DAOs and such, but I don't see a huge reason to make our code more complex now to account for a maybe. Don't feel super strongly though.

Reply:
Yep, that was my thinking as well.  For public-facing APIs it makes sense to try to be as future proof as possible, but for internal-only code I've found it's generally not worth it to try to code for features that don't exist yet as more often than not by the time you get around to actually implementing the feature you wind up having to redo everything anyway.

Reply:
The change is in the `context` which is public facing. Also, I think this change is really not complicating our code, to be honest. 🙂

Reply:
I actually don't think it is?  AFAIK, the main public-facing APIs that correspond primarily to CRUD operations on Documents (plus Ceramic.create() to get started) are:
* `Ceramic.create()`
* `Ceramic.createDocument()`
* `Ceramic.createDocumentFromGenesis()`
* `Ceramic.loadDocument()`
* `Document.change()`

None of those take Context.  I believe Context is an internal-only object for hanging process-wide globals off of, whose contents are derived from the `CeramicConfig` passed to `Ceramic.create()`.  I think we should strive to keep Context out of any APIs that we expect end user developers to use.  If it were to every wind up in a function that users use, it would hopefully only be as a black box that is passed through into lower-level functions.

Anyway back to the original convo: IMO any code written for a feature that doesn't exist yet is complication by definition.  But in this case it's certainly not a huge difference either way so if you feel strongly that we should store an array for each chainId then that's fine

Reply:
I think you're not looking at the right APIs. One such is a static method [makeGenesis](https://github.com/ceramicnetwork/js-ceramic/blob/f23ff938e92e2f835f613e8e3162fe7795194ebf/packages/common/src/doctype.ts#L210). On the last iteration that we had @oed we decided that we're going to put `preferredChain` in the `context`, right?

Also, every `Doctype` instance has a `context` instance by default.

Reply:
I hadn't imagined any end-user developer would ever call `TileDoctype.makeGenesis` directly.  I'd imagined they would call `CeramicApi.createDocumentFromGenesis` instead.

I do think this conversation illustrates that we should probably spend some time clarifying which of our APIs are designed to be used by developers building on Ceramic and which are only for developers *of* Ceramic.  I filed https://github.com/ceramicnetwork/js-ceramic/issues/554 for this